src/components/GitStash.tsx

```diff
@@ -0,0 +1,386 @@
+import { caretDownIcon, caretRightIcon } from '@jupyterlab/ui-components';
+import * as React from 'react';
+import {
+  changeStageButtonStyle,
+  sectionAreaStyle,
+  sectionFileContainerStyle,
+  sectionHeaderLabelStyle
+} from '../style/GitStageStyle';
+import { Git } from '../tokens';
+import { GitExtension } from '../model';
+import { hiddenButtonStyle } from '../style/ActionButtonStyle';
+// import { fileListWrapperClass } from '../style/FileListStyle';
+import { ActionButton } from './ActionButton';
+import { addIcon, discardIcon, removeIcon } from '../style/icons';
+import { TranslationBundle } from '@jupyterlab/translation';
+import { UseSignal } from '@jupyterlab/apputils';
+import { FixedSizeList } from 'react-window';
+import {
+  fileIconStyle,
+  fileLabelStyle,
+  folderLabelStyle
+} from '../style/FilePathStyle';
+import { fileIcon } from '@jupyterlab/ui-components';
+
+const HEADER_HEIGHT = 34;
+const ITEM_HEIGHT = 25;
+
+export interface IGitStashProps {
+  /**
+   * Actions component to display at the far right of the stage
+   */
+  actions?: React.ReactElement;
+
+  /**
+   * Is this group collapsible
+   */
+  collapsible?: boolean;
+
+  /**
+   * Git extension model
+   */
+  model: GitExtension;
+
+  /**
+   * Files in the group
+   */
+  stash: Git.IStash;
+  /**
+   * HTML element height
+   */
+  height: number;
+  /**
+   * Optional select all element
+   */
+  selectAllButton?: React.ReactElement;
+
+  /**
+   * The application language translator.
+   */
+  trans: TranslationBundle;
+
+  /**
+   * Wrap mouse event handler to stop event propagation
+   * @param fn Mouse event handler
+   * @returns Mouse event handler that stops event from propagating
+   */
+  stopPropagationWrapper: (
+    fn: React.EventHandler<React.MouseEvent>
+  ) => React.EventHandler<React.MouseEvent>;
```

Comment:
Is this needed?

Reply:
Yes, we need to wrap the Action Button with the stopPropagationWrapper. 

Without it, if we click an ActionButton, it will also pass the click event up to the parent, where it will trigger the collapsible.

```
     <ActionButton
                          className={hiddenButtonStyle}
                          icon={removeIcon}
                          title={'Apply stash entry (prop wrapper)'}
                          onClick={props.stopPropagationWrapper(() => {
                            gitStashApply(entry.index);
                          })}
                        />
```


With it, if we click an ActionButton, we will not trigger the parent collapsible.

```
  <ActionButton
                          className={hiddenButtonStyle}
                          icon={removeIcon}
                          title={'Apply stash entry (no prop wrapper)'}
                          onClick={() => {
                            gitStashApply(entry.index);
                          }}
                        />
```


https://user-images.githubusercontent.com/94336773/230666164-07bfbecd-a09a-4e85-a62d-b956717d5b2f.mov

Reply:
Makes sense. Thanks for explaining 😄